index.js

```diff
@@ -338,6 +340,7 @@ function asPromise(opts) {
 						throw new got.HTTPError(statusCode, res.statusMessage, res.headers, opts);
 					}
 
+					proxy.emit('response', res);
```

Comment:
Shouldn't this be called right (on line 319) away, like the stream version, instead of waiting for the body to be fetched?

Reply:
Yeah, you right let me remove it.